### DISCUSSION TITLE
モデルの自動読み込みについて

### DISCUSSION THREAD

**Discussion by nira-yusan** [pixiv/vroid-sdk-developers#116] [Q&A]
「一度読み込んだことのあるモデルを、ゲーム終了→ゲーム起動して続きからプレイ、といったことをしても自動で読み込む」という機能を実装したく、色々と調べているのですがいくつか聞きたいことがあります。

・キャッシュの保存などは規約違反とありますが、キャラクターモデルIDのゲームセーブデータへの保存およびそれを利用した自動読み込みは可能でしょうか？
・一度読み込んだことのあるモデルを自動読み込みする際、ライセンス表示などは必要でしょうか？

**Comment by nyamadan** [comment]
> 「一度読み込んだことのあるモデルを、ゲーム終了→ゲーム起動して続きからプレイ、といったことをしても自動で読み込む」という機能を実装したく、色々と調べているのですがいくつか聞きたいことがあります。

> ・キャッシュの保存などは規約違反とありますが、キャラクターモデルIDのゲームセーブデータへの保存およびそれを利用した自動読み込みは可能でしょうか？ 

ご指摘の通り原則モデルデータそのものをストレージに保存する行為は禁止となります。
キャラクターモデルIDをストレージなどに保存して次回読み込み時に利用することに関しては特に現時点では禁止しておりません。
ただし、VRoid SDKにはこれを実現する機能は存在しないため、実装いただく必要がございます。

> ・一度読み込んだことのあるモデルを自動読み込みする際、ライセンス表示などは必要でしょうか？

こちらにつきましては、ライセンスが変更されている可能性が存在するため、都度ライセンスを取得し、表示することをお勧めいたします。

**Reply by nira-yusan** [reply]
なるほど、モデルIDの取得方法を教えてもらえませんか？

**Reply by nyamadan** [reply]
[DefaultApi.GetAccountCharacterModels](https://developer.vroid.com/sdk/docs/0.3.0/Pixiv.VroidSdk.Api.DefaultApi.html#Pixiv_VroidSdk_Api_DefaultApi_GetAccountCharacterModels_System_Int32_System_Action_System_Collections_Generic_List_Pixiv_VroidSdk_Api_DataModel_CharacterModel___System_Action_Pixiv_VroidSdk_Api_DataModel_ApiErrorFormat__) などから取得できる [CharacterModel](https://developer.vroid.com/sdk/docs/0.3.0/Pixiv.VroidSdk.Api.DataModel.CharacterModel.html) クラスの [id](https://developer.vroid.com/sdk/docs/0.3.0/Pixiv.VroidSdk.Api.DataModel.CharacterModel.html#Pixiv_VroidSdk_Api_DataModel_CharacterModel_id) フィールドから取得可能となっています。
